vnext/Microsoft.ReactNative/JsiApi.idl

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.ReactNative
+{
+  interface IJsiRuntime;
+
+  delegate void JsiDataHandler(UInt8[] data);
+
+  [webhosthidden]
+  interface IJsiBuffer
+  {
+    UInt32 Size { get; };
+    void GetData(JsiDataHandler handler);
+  };
+
+  enum JsiValueKind
+  {
+    Undefined,
+    Null,
+    Boolean,
+    Number,
+    Symbol,
+    String,
+    Object,
+  };
+
+  [webhosthidden]
+  struct JsiValueData
+  {
+    JsiValueKind Kind;
+    UInt64 Data;
+  };
+
+  struct JsiSymbolData
+  {
+    UInt64 Data;
+  };
+
+  struct JsiStringData
```

Comment:
>JsiStringData [](start = 9, length = 13)

So, do we need all these identical structures defined in .idl file? Looks like this could be just "PointerData", and then maybe have typed wrappers in .h file?

Reply:
This is what I started with. They are implemented to have better type checking for the parameters. Though, I may reconsider it...

---
In reply to: [455234225](https://github.com/microsoft/react-native-windows/pull/5461#discussion_r455234225) [](ancestors = 455234225)